umbraco/Umbraco-CMS
# V13: Add eventype to webhookevents

Fixes https://github.com/umbraco/Umbraco-CMS/issues/15095

# Notes
- Adds new enum `WebhookEventType`.
- Adds a `EventType` property to IWebhookEvent.
- Refactors front end to now take into account the new EventType 🙌 
- This should now also fix up the filtering of events, so we now no longer filter by event name, but type instead.

# How to test
- Go to the Settings -> Webhook section
- Create a webhook with some events
- Assert when selecting event that it filters properly (When you choose a content event, it should filter the media ones)
- Assert that if you choose to re-pick the events, it remembers which one you already have chosen.
- Save the webhook.
- Assert you can pick other events and re-save
